internal/pkg/handler/updated-handler.go

```diff
@@ -1,8 +1,23 @@
 package handler
 
 import (
+	"bytes"
```

Comment:
Is `updated-controller` a  good name for this? :thinking:  @faizanahmad055 @waseem-h